internal/pkg/test/privilege_linux.go

```diff
@@ -21,10 +21,10 @@ var origUID, origGID, unprivUID, unprivGID int
 var origHome, unprivHome string
 
 const (
-	// CacheDirPriv is the directory the cachedir gets set to when running privileged
-	CacheDirPriv = "/tmp/WithPrivilege"
-	// CacheDirUnpriv is the directory the cachedir gets set to when running unprivileged
-	CacheDirUnpriv = "/tmp/WithoutPrivilege"
+	// BaseDirPriv is the directory used by tests when running privileged
+	BaseDirPriv = "/tmp/WithPrivilege"
+	// BaseDirUnpriv is the directory used by tests when running unprivileged
+	BaseDirUnpriv = "/tmp/WithoutPrivilege"
```

Comment:
We're implicitly making the assumption here that `/tmp` is accessible with and without privilege. Since this is the case, let's create the temporary cache dirs directly in `/tmp`, which removes the need for the state that is being shared in `SINGULARITY_TESTBASEDIR`, and removes the coupling between privilege and cachedir handling.

Reply:
Will do.

Reply:
BTW, I misunderstood at first what you had in mind, I think we are now on the same page, I believe this is resolved.